xbmc/video/tags/VideoTagLoaderPlugin.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+/*
+ *      Copyright (C) 2005-2017 Team XBMC
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "IVideoInfoTagLoader.h"
+#include "video/VideoInfoTag.h"
+
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+//! \brief Video tag loader from plugin source.
+class CVideoTagLoaderPlugin : public VIDEO::IVideoInfoTagLoader
+{
+public:
+  CVideoTagLoaderPlugin(const CFileItem& item, bool forceRefresh);
+
+  virtual ~CVideoTagLoaderPlugin() = default;
+
+  //! \brief Returns whether or not read has info.
+  bool HasInfo() const override;
+
+  //! \brief Load "tag" from plugin.
+  //! \brief tag Tag to load info into
+  CInfoScanner::INFO_TYPE Load(CVideoInfoTag& tag, bool prioritise,
+                               std::vector<EmbeddedArt>* = nullptr) override;
+
+  std::unique_ptr<std::map<std::string, std::string>> m_art;
```

Comment:
Please don't make member variables public, especially not ones starting with `m_`

Reply:
I've made this the same as in e.g. CFileItem. There is no need to make this private and add public "get" method. This variable is set only in this class and used only in CVideoLibraryRefreshingJob (std::move to local variable).

Anyway I'll fix it. This is not a matter of principle.

Reply:
> I've made this the same as in e.g. CFileItem

Kodi is an old project with lots of anti-patterns. Assume that you can do a better job than what we currently have, and don't be afraid to change existing code.

> There is no need to make this private and add public "get" method

That is exactly what you should do:

```c++
public:
  std::map<std::string, std::string> &GetArt() { return m_art; }

private:
  std::map<std::string, std::string> m_art;
```

Of course, it sounds like you have a better idea that you should pursue.

std::move works with std::map, so is std::unique_ptr necessary?

Reply:
Because there can be a valid empty std::map (no art), so that std::unique_ptr is used to indicate, that art is valid. This is a universal solution to show whether we can use this art map. I can add additional boolean property and method "bool HasArt()" if you don't like unique_ptr.

Reply:
Gotchya. Your unique_ptr is fine.